currency.py

```diff
@@ -8,71 +8,72 @@ def EXIT(message, exit_code):
     sys.exit(exit_code)
 
 
-def date_check(date):    #we just need to check if date in range [1992-07-01;<today>] dmy
+def date_check(dateinp):
+    min_date = datetime(1992, 7, 1)
+    max_date = datetime.today()
 
-    lst[0] = int(lst[0])
-    lst[1] = int(lst[1])
-    lst[2] = int(lst[2])
-    today = datetime.today()
-
-    if date[0] > today.day and date[2] == today.year and date[1] == today.month:
-        return False
-
-    if (date[1] > today.month and date[2] == today.year) or (date[2] == 1992 and date[1] < 7):
-        return False
-
-    if date[2] < 1992 or date[2] > today.year:
-        return False
-
-    return True
+    if min_date <= dateinp <= max_date:
```

Comment:
No need for the `if` statement here. You can do this instead:

```python
return datetime(1992, 7, 1) <= dateinp <= datetime.today()
```

because the comparison operators already return the boolean values. For instance:

```python
a = 1 > 2
print(a) # Will print 'False', because `a` is Bool
```